lib/buffer.js

```diff
@@ -1141,10 +1142,10 @@ Buffer.prototype.write = function write(string, offset, length, encoding) {
 
 Buffer.prototype.toJSON = function toJSON() {
   if (this.length > 0) {
-    const data = new Array(this.length);
-    for (let i = 0; i < this.length; ++i)
-      data[i] = this[i];
-    return { type: 'Buffer', data };
+    return {
+      type: 'Buffer',
+      data: ArrayFrom(new Array(this.length), (_value, i) => this[i]),
```

Comment:
Would `ArrayOf(...this)` work?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/of